crates/hir_ty/src/display.rs

```diff
@@ -539,6 +539,11 @@ impl HirDisplay for Ty {
                 }
             }
             TyKind::Closure(.., substs) => {
+                if f.display_target.is_source_code() {
+                    return Err(HirDisplayError::DisplaySourceCodeError(
+                        DisplaySourceCodeError::UnknownType,
```

Comment:
We probably want a new variant here like `DisplaySourceCodeError::Closure`

Reply:
IMO we should display closures as `impl Fn` types, then we wouldn't need the error.

Reply:
> We probably want a new variant here like `DisplaySourceCodeError::Closure`

Added.

Reply:
Right that totally slipped my mind, that is probably even better. Though we can't know currently what `Fn*` type we would need since our closures don't track what they capture right?

Reply:
Yeah, that requires some analysis we can't do yet.